### ISSUE TITLE
Broken Internet connection if app suddenly stopped

### ISSUE THREAD

**Issue by monkeber** [OpenVPN/openvpn3#105]
I'm using openvpn3 (3.4.1-1) library installed from vcpkg on Windows 10, for my code I have derived a class from `ClientAPI::OpenVPNClient`, for establishing connection it works well but there is a big problem: when I stop debugging or end my app via Task Manager (so basically when I stop application without disconnecting OpenVPN first) and some OpenVPN connection was established - the computer that the app was running on loses any Internet connectivity and the only way to get it back which I found so far is to restore Windows from restoration point. I have tried to enable and disable adapters, restart computer, release and renew routing tables - doesn't help. Network settings seems to be as they were before connection (i.e. my network listed as operational, etc.), routing tables also as they were before.

Am I missing something with this or is this supposed behavior? What the library does to the connection so I can enable connectivity back?

**Comment by lstipakov** [comment]
Hi,

At least DNS settings are not restored if app is force-stopped, but we are not aware of any other consequences.

Could you you reproduce this issue with official test client (https://ci.appveyor.com/project/openvpn/openvpn3/build/artifacts, use ovpncli.exe)? 

What comes to broken DNS - you might want to remove registry keys which names start with "OpenVPNDNSRouting" at path HKLM\\SYSTEM\\CurrentControlSet\\Services\\Dnscache\\Parameters\\DnsPolicyConfig.

**Comment by monkeber** [comment]
Hi, thanks for the fast reply, I will try it this weekend and get back to you.

**Comment by monkeber** [comment]
@lstipakov I tried your suggestion and yes, the problem were registry keys, after I deleted them everything works again.

I also tried to reproduce this issue on test client - unsuccessfully, after killing terminal via Task Manager the connection was still up and running, as well as Tap interface in the list of my adapters was listed as Connected, in my app it was disconnected after sudden stop.

Anyway thank for your help very much, I think this issue could be closed.
